### PR TITLE
React to aspnet/Razor#268 changes

### PR DESCRIPTION
**Pull request by dougbu**
- address #1253 using new Razor capabilities
- update baselines to match latest code generation
- use new `CodeBuilderContext` and `TagHelperAttributeValueCodeRenderer` signatures
- test complex expressions in bound non-string attribute values

### CODE REVIEW

**Comment by dougbu** [comment]
/cc @NTaylorMullen

about the only thing I might add is a few tests of more complicated attribute values (e.g. add another property or two to `InputTestTagHelper` and use that/them in ModelExpressionTagHelper.cshtml and the `MvcRazorHostTest` tests). playing with this but may punt because it would duplicate tests in the Razor repo

File: test/Microsoft.AspNet.Mvc.FunctionalTests/Compiler/Resources/MvcTagHelpersWebSite.MvcTagHelper_Home.Order.html [hunk 1]
```diff
@@ -18,6 +18,7 @@
         </div>
         <div>
             <label class="order" for="Products">Products</label>
+
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/Compiler/Resources/MvcTagHelpersWebSite.MvcTagHelper_Home.Order.html]: ?

**Comment by NTaylorMullen** [comment]
:shipit:

File: test/Microsoft.AspNet.Mvc.FunctionalTests/Compiler/Resources/MvcTagHelpersWebSite.MvcTagHelper_Home.Order.html (hunk 1 above)
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/Compiler/Resources/MvcTagHelpersWebSite.MvcTagHelper_Home.Order.html]: added a c# statement line to the corresponding .cshtml file (last file on this page)

Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/Compiler/Resources/MvcTagHelpersWebSite.MvcTagHelper_Home.Order.html]: :+1:

**Comment by dougbu** [comment]
@pranavkm second commit updated.

File: test/Microsoft.AspNet.Mvc.Razor.Host.Test/InjectChunkVisitorTest.cs [hunk 2]
```diff
@@ -149,7 +150,8 @@ private static CodeBuilderContext CreateContext()
                                          "MyClass",
                                          "MyNamespace",
                                          string.Empty,
-                                         shouldGenerateLinePragmas: true));
+                                         shouldGenerateLinePragmas: true),
+                new ParserErrorSink());
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.Razor.Host.Test/InjectChunkVisitorTest.cs]: indentation

File: test/Microsoft.AspNet.Mvc.Razor.Host.Test/ModelChunkVisitorTest.cs [hunk 3]
```diff
@@ -105,7 +106,8 @@ private static CodeBuilderContext CreateContext()
                                          "MyClass",
                                          "MyNamespace",
                                          string.Empty,
-                                         shouldGenerateLinePragmas: true));
+                                         shouldGenerateLinePragmas: true),
+                new ParserErrorSink());
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.Razor.Host.Test/ModelChunkVisitorTest.cs]: indent.

File: test/WebSites/MvcTagHelpersWebSite/Views/MvcTagHelper_Home/ProductList.cshtml [hunk 4]
```diff
@@ -10,9 +10,17 @@
 <body>
     @using (@Html.BeginForm("Index", "MvcTagHelper_Product", FormMethod.Post))
     {
-        @foreach (var product in Model)
+        var index = 0;
+        var fieldPrefix = ViewData.TemplateInfo.HtmlFieldPrefix;
+        foreach (var model in Model)
         {
-            @await Html.PartialAsync("_ProductPartial", product)
+            ViewData.TemplateInfo.HtmlFieldPrefix = fieldPrefix + string.Format("[{0}]", index++);
```
Comment (pranavkm) [inline test/WebSites/MvcTagHelpersWebSite/Views/MvcTagHelper_Home/ProductList.cshtml]: Perhaps write a Razor comment here explaining why this is done. It's not the most obvious of pieces.

**Comment by pranavkm** [comment]
:shipit: once comments are addressed.

File: test/Microsoft.AspNet.Mvc.Razor.Host.Test/ModelChunkVisitorTest.cs (hunk 3 above)
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Razor.Host.Test/ModelChunkVisitorTest.cs]: :-1: this is the second parameter to to the `CodeGeneratorContext` constructor, not an additional parameter to the `MvcRazorHost` constructor

File: test/Microsoft.AspNet.Mvc.Razor.Host.Test/InjectChunkVisitorTest.cs (hunk 2 above)
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Razor.Host.Test/InjectChunkVisitorTest.cs]: :-1: this is the second parameter to to the `CodeGeneratorContext` constructor, not an additional parameter to the `MvcRazorHost` constructor

File: test/WebSites/MvcTagHelpersWebSite/Views/MvcTagHelper_Home/ProductList.cshtml (hunk 4 above)
Comment (dougbu) [inline test/WebSites/MvcTagHelpersWebSite/Views/MvcTagHelper_Home/ProductList.cshtml]: :ok_hand:

**Comment by dougbu** [comment]
Commits a77d0718301cd1be5e3900e21167768c7f964d8d and d2fe1ebad763c948ecc47612d01c26a7eff9a1f6

**Pull request closed without merging**